### PR TITLE
feat(Forms): add `autoPushWhen` property to Iterate.PushContainer

### PR DESCRIPTION
**Pull request by tujoworker**
This is an addition to #3843

Quick example usage:

```tsx
<Iterate.Array path="/countries">
  <MyViewItem />
  <MyEditItem /> // Contains <Field.SelectCountry />
</Iterate.Array>

<Iterate.PushContainer
  path="/countries"
  autoPushWhen={(list) => list.length === 0} // define if and when it should push
>
  <MyEditItem />
</Iterate.PushContainer>
```

### CODE REVIEW

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/PushContainer/demos.mdx [hunk 1]
```diff
@@ -14,3 +14,9 @@ import * as Examples from './Examples'
 ### With existing data
 
 <Examples.ViewAndEditContainer />
+
+### Auto push
+
+This example makes use of the `autoPushWhen` prop. It automatically pushes a new item when a field inside the container is changed. You can use this feature only with a single toggle-able field.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/PushContainer/demos.mdx:20]:
```suggestion
This example makes use of the `autoPushWhen` prop. It automatically pushes a new item when a field inside the container is changed. You can use this feature only with a single toggleable field.
```

File: packages/dnb-eufemia/src/extensions/forms/Iterate/PushContainer/PushContainerDocs.ts [hunk 2]
```diff
@@ -21,6 +21,16 @@ export const PushContainerProperties: PropertiesTableProps = {
     type: 'function',
     status: 'optional',
   },
+  autoPushWhen: {
+    doc: 'Define if the container should auto save when e.g. the first item (list is empty). Should be a function that returns a boolean.',
```
Comment (langz) [inline packages/dnb-eufemia/src/extensions/forms/Iterate/PushContainer/PushContainerDocs.ts:25]:
```suggestion
    doc: 'Define if the container should auto push when e.g. the first item (list is empty). Should be a function that returns a boolean.',
```

File: packages/dnb-eufemia/src/extensions/forms/Iterate/PushContainer/PushContainer.tsx [hunk 3]
```diff
@@ -32,11 +33,22 @@ export type Props = {
    */
   showOpenButtonWhen?: (list: unknown[]) => boolean
 
+  /**
+   * Define if the container should auto save when e.g. the first item (list is empty).
```
Comment (langz) [inline packages/dnb-eufemia/src/extensions/forms/Iterate/PushContainer/PushContainer.tsx:37]:
```suggestion
   * Define if the container should auto push when e.g. the first item (list is empty).
```

**Review by langz — APPROVED** [review]
👏 

Should the property name be `autoPushWhen` or `pushWhen`?
Does the word auto(automatically) add any meaning?

I don't have any strong opinions here, just wanted to give it a thought.

**Comment by tujoworker** [comment]
Closing this in favor of #3877

**Pull request closed without merging**